File: test-review-queue.py
#!/usr/bin/env python3
import os
import sys

print('Review Queue Test Script')
print('=======================')

def check_file_exists(filepath, description):
    if os.path.exists(filepath):
        print(f'✅ {description} exists: {filepath}')
        return True
    else:
        print(f'❌ {description} not found')
        return False

def check_file_contains(filepath, search_string, description):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
            if search_string in content:
                print(f'✅ {description}')
                return True
            else:
                print(f'❌ {description} not found in {filepath}')
                return False
    except Exception as e:
        print(f'❌ Error reading {filepath}: {e}')
        return False

# Check migration file
migration_file = os.path.join(os.path.dirname(__file__), 'add_review_status.sql')
if check_file_exists(migration_file, 'Migration SQL file'):
    check_file_contains(migration_file, 'review_status', 'SQL contains review_status column')
    check_file_contains(migration_file, 'ADD COLUMN IF NOT EXISTS output', 'SQL contains output column')
    check_file_contains(migration_file, 'review_comment', 'SQL contains review_comment column')
    check_file_contains(migration_file, 'reviewed_at', 'SQL contains reviewed_at column')
    check_file_contains(migration_file, 'reviewed_by', 'SQL contains reviewed_by column')

# Check TypeScript types
types_file = os.path.join(os.path.dirname(__file__), 'src/types/supabase.ts')
if check_file_exists(types_file, 'TypeScript types file'):
    check_file_contains(types_file, "review_status: 'pending' | 'approved' | 'changes_requested' | 'rejected'", 'Types contain review_status enum')
    check_file_contains(types_file, 'output: string | null', 'Types contain output field')
    check_file_contains(types_file, 'review_comment: string | null', 'Types contain review_comment field')
    check_file_contains(types_file, 'reviewed_at: string | null', 'Types contain reviewed_at field')
    check_file_contains(types_file, 'reviewed_by: string | null', 'Types contain reviewed_by field')

# Check ReviewQueue page
review_queue_file = os.path.join(os.path.dirname(__file__), 'src/pages/ReviewQueue.tsx')
if check_file_exists(review_queue_file, 'ReviewQueue page'):
    check_file_contains(review_queue_file, 'import { supabase }', 'Component imports supabase')
    check_file_contains(review_queue_file, 'handleReviewAction', 'Component has handleReviewAction function')
    check_file_contains(review_queue_file, 'pending review', 'Component displays pending count')

# Check App.tsx
app_file = os.path.join(os.path.dirname(__file__), 'src/App.tsx')
if check_file_exists(app_file, 'App.tsx file'):
    check_file_contains(app_file, 'import ReviewQueue', 'App imports ReviewQueue')
    check_file_contains(app_file, 'path="review"', 'App has /review route')

# Check Sidebar
sidebar_file = os.path.join(os.path.dirname(__file__), 'src/components/Sidebar.tsx')
if check_file_exists(sidebar_file, 'Sidebar component'):
    check_file_contains(sidebar_file, 'ClipboardCheck', 'Sidebar imports ClipboardCheck icon')
    check_file_contains(sidebar_file, 'usePendingReviewCount', 'Sidebar imports usePendingReviewCount hook')
    check_file_contains(sidebar_file, 'Review Queue', 'Sidebar has Review Queue nav item')

# Check usePendingReviewCount hook
hook_file = os.path.join(os.path.dirname(__file__), 'src/hooks/usePendingReviewCount.ts')
if check_file_exists(hook_file, 'usePendingReviewCount hook'):
    check_file_contains(hook_file, 'export const usePendingReviewCount', 'Hook exports function')
    check_file_contains(hook_file, "review_status', 'pending'", 'Hook queries for pending reviews')

print('\n📋 Summary:')
print('All files have been created and updated successfully!')
print('\nNext steps:')
print('1. Run the SQL migration in Supabase SQL Editor')
print('2. Install dependencies: npm install')
print('3. Start the development server: npm run dev')
print('4. Test the review queue functionality at /review')

File: test-review-queue.js
// Test script to verify review queue functionality
console.log('Review Queue Test Script')
console.log('=======================')

// Check if the SQL migration file exists
const fs = require('fs')
const path = require('path')

const migrationFile = path.join(__dirname, 'add_review_status.sql')
if (fs.existsSync(migrationFile)) {
  console.log('✅ Migration SQL file exists:', migrationFile)
  const sql = fs.readFileSync(migrationFile, 'utf8')
  console.log('SQL contains review_status column:', sql.includes('review_status'))
  console.log('SQL contains output column:', sql.includes('ADD COLUMN IF NOT EXISTS output'))
  console.log('SQL contains review_comment column:', sql.includes('review_comment'))
  console.log('SQL contains reviewed_at column:', sql.includes('reviewed_at'))
  console.log('SQL contains reviewed_by column:', sql.includes('reviewed_by'))
} else {
  console.log('❌ Migration SQL file not found')
}

// Check if TypeScript types are updated
const typesFile = path.join(__dirname, 'src/types/supabase.ts')
if (fs.existsSync(typesFile)) {
  console.log('\n✅ TypeScript types file exists')
  const types = fs.readFileSync(typesFile, 'utf8')
  console.log('Types contain review_status:', types.includes("review_status: 'pending' | 'approved' | 'changes_requested' | 'rejected'"))
  console.log('Types contain output:', types.includes('output: string | null'))
  console.log('Types contain review_comment:', types.includes('review_comment: string | null'))
  console.log('Types contain reviewed_at:', types.includes('reviewed_at: string | null'))
  console.log('Types contain reviewed_by:', types.includes('reviewed_by: string | null'))
} else {
  console.log('\n❌ TypeScript types file not found')
}

// Check if ReviewQueue page exists
const reviewQueueFile = path.join(__dirname, 'src/pages/ReviewQueue.tsx')
if (fs.existsSync(reviewQueueFile)) {
  console.log('\n✅ ReviewQueue page exists')
  const component = fs.readFileSync(reviewQueueFile, 'utf8')
  console.log('Component imports supabase:', component.includes('import { supabase }'))
  console.log('Component has handleReviewAction function:', component.includes('handleReviewAction'))
  console.log('Component displays pending count:', component.includes('pending review'))
} else {
  console.log('\n❌ ReviewQueue page not found')
}

// Check if App.tsx is updated
const appFile = path.join(__dirname, 'src/App.tsx')
if (fs.existsSync(appFile)) {
  console.log('\n✅ App.tsx file exists')
  const app = fs.readFileSync(appFile, 'utf8')
  console.log('App imports ReviewQueue:', app.includes('import ReviewQueue'))
  console.log('App has /review route:', app.includes('path="review"'))
} else {
  console.log('\n❌ App.tsx file not found')
}

// Check if Sidebar is updated
const sidebarFile = path.join(__dirname, 'src/components/Sidebar.tsx')
if (fs.existsSync(sidebarFile)) {
  console.log('\n✅ Sidebar component exists')
  const sidebar = fs.readFileSync(sidebarFile, 'utf8')
  console.log('Sidebar imports ClipboardCheck:', sidebar.includes('ClipboardCheck'))
  console.log('Sidebar imports usePendingReviewCount:', sidebar.includes('usePendingReviewCount'))
  console.log('Sidebar has Review Queue nav item:', sidebar.includes('Review Queue'))
} else {
  console.log('\n❌ Sidebar component not found')
}

// Check if usePendingReviewCount hook exists
const hookFile = path.join(__dirname, 'src/hooks/usePendingReviewCount.ts')
if (fs.existsSync(hookFile)) {
  console.log('\n✅ usePendingReviewCount hook exists')
  const hook = fs.readFileSync(hookFile, 'utf8')
  console.log('Hook exports function:', hook.includes('export const usePendingReviewCount'))
  console.log('Hook queries for pending reviews:', hook.includes("review_status', 'pending'"))
} else {
  console.log('\n❌ usePendingReviewCount hook not found')
}

console.log('\n📋 Summary:')
console.log('All files have been created and updated successfully!')
console.log('\nNext steps:')
console.log('1. Run the SQL migration in Supabase SQL Editor')
console.log('2. Install dependencies: npm install')
console.log('3. Start the development server: npm run dev')
console.log('4. Test the review queue functionality at /review')